### ISSUE TITLE
[Drupal] Upgrade from 7 to 8 everything that can be done

### ISSUE THREAD

**Issue by TomasVotruba** [rectorphp/rector#1604]
@mxr576 What do you think? Any chance you can give me some clues or help with this?

**Comment by mxr576** [comment]
TL;DR I do not think it is a good idea to suggest anybody to try to do an automated code update from Drupal 7 to Drupal 8 :) The outcome could be something that works(tm) but 98% it won't be an optimal solution.

Drupal 7 is purely procedural. Drupal 8 introduced OOP, Symfony, service container, events, etc. Also, some things that used to be defined in PHP code moved to YAML files, for example, menu definitions: 
https://www.drupal.org/docs/8/converting-drupal-7-modules-to-drupal-8/d7-to-d8-upgrade-tutorial-convert-hook_menu-and-hook As far as I know, Rector does not create about YAML files. :)

Partially this is the reason why I call my library Drupal8-rector and not Drupal-rector ;)

**Comment by TomasVotruba** [comment]
> Drupal 7 is purely procedural. Drupal 8 introduced OOP, Symfony, service container, events, etc. 

Isn't that the main goal of Rector? To help with the hardest migrations like this?

I mean it's nice to upgrade from PHP 7.0 to PHP 7.1, but that can be done by programmer, even if it take 5 more days.

But how many webs are still running on Drupal 7?
And how many of them will still be active in next 2-3 years and will need to migrate to never Drupal?

And how many happy programmers there would be, if Rector would handle the PHP part for them? :)

**Comment by TomasVotruba** [comment]
Related https://www.drupal.org/docs/8/converting-drupal-7-modules-to-drupal-8

**Comment by TomasVotruba** [comment]
Closing as out of scope at the moment
